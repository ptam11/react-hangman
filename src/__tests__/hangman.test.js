import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import App from '../App';
import Hangman from '../Hangman';

import img0 from '../../src/0.jpg';
import img1 from '../../src/1.jpg';

configure({ adapter: new Adapter() });

it('App renders without crashing', function() {
	shallow(<App />);
});

it('Hangman Component renders without errors', function() {
	const div = document.createElement('div');
	ReactDOM.render(<Hangman />, div);
	ReactDOM.unmountComponentAtNode(div);
});

it('matches snapshot', function() {
	let wrapper = shallow(<Hangman />);
	let serialized = toJson(wrapper);
	expect(serialized).toMatchSnapshot();
});

it('it should should not change image on correct guess', function() {
  let wrapper = mount(<Hangman />);
  wrapper.setState({ 'answer': 'apple' })
	wrapper.find("button[value='a']").simulate('click', { target: { value: 'a' } });

	expect(wrapper.html()).toContain(img0);
});

it('it should render img1 after incorrect guess', function() {
	let wrapper = mount(<Hangman />);
	wrapper.setState({ answer: 'apple' });
	wrapper.find("button[value='z']").simulate('click', { target: { value: 'z' } });

	expect(wrapper.html()).toContain(img1);
});

it('it should update number of wrong guesses', function() {
	let wrapper = mount(<Hangman />);
	wrapper.setState({ answer: 'apple' });
	wrapper.find("button[value='z']").simulate('click', { target: { value: 'z' } });

	expect(wrapper.state('nWrong')).toEqual(1);
	// expect(wrapper.exists('a')).toEqual(true);
	// expect(wrapper.html()).toContain("Think of a Question");
});

it('it should contain an image after maxGuesses', function() {
	let wrapper = mount(<Hangman />);
	wrapper.setState({ answer: 'apple' });
	// simulate maxGuesses
	wrapper.find("button[value='z']").simulate('click', { target: { value: 'z' } });
	wrapper.find("button[value='y']").simulate('click', { target: { value: 'y' } });
	wrapper.find("button[value='w']").simulate('click', { target: { value: 'w' } });
	wrapper.find("button[value='v']").simulate('click', { target: { value: 'v' } });
	wrapper.find("button[value='t']").simulate('click', { target: { value: 't' } });
	wrapper.find("button[value='u']").simulate('click', { target: { value: 'u' } });

	expect(wrapper.html()).toContain('.jpg');
});

it('it should not allow clicks after maxGuesses', function() {
	let wrapper = mount(<Hangman />);
	wrapper.setState({ answer: 'apple' });
	// simulate maxGuesses
	wrapper.find("button[value='z']").simulate('click', { target: { value: 'z' } });
	wrapper.find("button[value='y']").simulate('click', { target: { value: 'y' } });
	wrapper.find("button[value='w']").simulate('click', { target: { value: 'w' } });
	wrapper.find("button[value='v']").simulate('click', { target: { value: 'v' } });
	wrapper.find("button[value='t']").simulate('click', { target: { value: 't' } });
	wrapper.find("button[value='u']").simulate('click', { target: { value: 'u' } });

  // searches for className
	expect(wrapper.html()).not.toContain('Hangman-ltr-btn');
});

// it('renders correct message', function() {
//   let wrapper = mount(<EightBall />);
//   wrapper.setState({text: "Yes changing", color: "blue"});

//   expect(wrapper.exists('p')).toEqual(true);
//   expect(wrapper.html()).toContain("Yes changing");
//   expect(wrapper.html()).toContain("blue");

// });

// it('changes default message on click', function() {
//   let wrapper = mount(<EightBall />);
//   wrapper.find('#EightBall').simulate("click");

//   expect(wrapper.exists('p')).toEqual(true);
//   expect(wrapper.html()).not.toContain("Think of a Question");

// });

// it('changes message to thinking', function() {
//   let wrapper = mount(<EightBall />);

//   wrapper.setState({answers: [
//     { msg: "It is certain.", color: "green" },
//     { msg: "It is decidedly so.", color: "green" },
//   ]});

//   wrapper.find('#EightBall').simulate("click");
//   expect(wrapper.exists('p')).toEqual(true);

//   // Let the ball think for 1s
//     expect(wrapper.find("#message").html()).toContain("Thinking");

// });

// it('works with random messages', function() {
//   let wrapper = mount(<EightBall />);

//   wrapper.setState({answers: [
//     { msg: "It is certain.", color: "green" },
//     { msg: "It is decidedly so.", color: "green" },
//   ]});

//   wrapper.find('#EightBall').simulate("click");
//   expect(wrapper.exists('p')).toEqual(true);

//   // Let the ball think for 1s
//   setTimeout( () => {
//     expect(wrapper.find("#message").html()).toContain("It is");
//   }, 1000);

// });

// it('resets the ball to Question', function() {
//   let wrapper = mount(<EightBall />);

//   wrapper.setState({answers: [
//     { msg: "It is certain.", color: "green" },
//     { msg: "It is decidedly so.", color: "green" },
//   ]});

//   wrapper.find('#reset').simulate("click");
//   expect(wrapper.exists('p')).toEqual(true);

//   expect(wrapper.find("#message").html()).toContain("Think of a Question");

// });
