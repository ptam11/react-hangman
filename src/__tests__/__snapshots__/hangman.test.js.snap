// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="Hangman"
>
  <p>
    Wrong guesses: 
    0
  </p>
  <img
    src="0.jpg"
  />
  <p
    className="Hangman-word"
  >
    _
    _
    _
    _
    _
  </p>
  <p>
    <button
      className="Hangman-ltr-btn"
      disabled={false}
      key="a"
      onClick={[Function]}
      value="a"
    >
      a
    </button>
    <button
      className="Hangman-ltr-btn"
      disabled={false}
      key="b"
      onClick={[Function]}
      value="b"
    >
      b
    </button>
    <button
      className="Hangman-ltr-btn"
      disabled={false}
      key="c"
      onClick={[Function]}
      value="c"
    >
      c
    </button>
    <button
      className="Hangman-ltr-btn"
      disabled={false}
      key="d"
      onClick={[Function]}
      value="d"
    >
      d
    </button>
    <button
      className="Hangman-ltr-btn"
      disabled={false}
      key="e"
      onClick={[Function]}
      value="e"
    >
      e
    </button>
    <button
      className="Hangman-ltr-btn"
      disabled={false}
      key="f"
      onClick={[Function]}
      value="f"
    >
      f
    </button>
    <button
      className="Hangman-ltr-btn"
      disabled={false}
      key="g"
      onClick={[Function]}
      value="g"
    >
      g
    </button>
    <button
      className="Hangman-ltr-btn"
      disabled={false}
      key="h"
      onClick={[Function]}
      value="h"
    >
      h
    </button>
    <button
      className="Hangman-ltr-btn"
      disabled={false}
      key="i"
      onClick={[Function]}
      value="i"
    >
      i
    </button>
    <button
      className="Hangman-ltr-btn"
      disabled={false}
      key="j"
      onClick={[Function]}
      value="j"
    >
      j
    </button>
    <button
      className="Hangman-ltr-btn"
      disabled={false}
      key="k"
      onClick={[Function]}
      value="k"
    >
      k
    </button>
    <button
      className="Hangman-ltr-btn"
      disabled={false}
      key="l"
      onClick={[Function]}
      value="l"
    >
      l
    </button>
    <button
      className="Hangman-ltr-btn"
      disabled={false}
      key="m"
      onClick={[Function]}
      value="m"
    >
      m
    </button>
    <button
      className="Hangman-ltr-btn"
      disabled={false}
      key="n"
      onClick={[Function]}
      value="n"
    >
      n
    </button>
    <button
      className="Hangman-ltr-btn"
      disabled={false}
      key="o"
      onClick={[Function]}
      value="o"
    >
      o
    </button>
    <button
      className="Hangman-ltr-btn"
      disabled={false}
      key="p"
      onClick={[Function]}
      value="p"
    >
      p
    </button>
    <button
      className="Hangman-ltr-btn"
      disabled={false}
      key="q"
      onClick={[Function]}
      value="q"
    >
      q
    </button>
    <button
      className="Hangman-ltr-btn"
      disabled={false}
      key="r"
      onClick={[Function]}
      value="r"
    >
      r
    </button>
    <button
      className="Hangman-ltr-btn"
      disabled={false}
      key="s"
      onClick={[Function]}
      value="s"
    >
      s
    </button>
    <button
      className="Hangman-ltr-btn"
      disabled={false}
      key="t"
      onClick={[Function]}
      value="t"
    >
      t
    </button>
    <button
      className="Hangman-ltr-btn"
      disabled={false}
      key="u"
      onClick={[Function]}
      value="u"
    >
      u
    </button>
    <button
      className="Hangman-ltr-btn"
      disabled={false}
      key="v"
      onClick={[Function]}
      value="v"
    >
      v
    </button>
    <button
      className="Hangman-ltr-btn"
      disabled={false}
      key="w"
      onClick={[Function]}
      value="w"
    >
      w
    </button>
    <button
      className="Hangman-ltr-btn"
      disabled={false}
      key="x"
      onClick={[Function]}
      value="x"
    >
      x
    </button>
    <button
      className="Hangman-ltr-btn"
      disabled={false}
      key="y"
      onClick={[Function]}
      value="y"
    >
      y
    </button>
    <button
      className="Hangman-ltr-btn"
      disabled={false}
      key="z"
      onClick={[Function]}
      value="z"
    >
      z
    </button>
  </p>
  <button
    className="Hangman-reset"
    onClick={[Function]}
  >
     Restart 
  </button>
</div>
`;
